sympy/polys/fields.py

```diff
@@ -493,9 +489,9 @@ def __mul__(f, g):
 
         if not f or not g:
             return field.zero
-        elif isinstance(g, field.dtype):
+        elif isinstance(g, FracElement) and g.field == field:
             return f.new(f.numer*g.numer, f.denom*g.denom)
-        elif isinstance(g, field.ring.dtype):
+        elif isinstance(g, PolyElement) and g.ring == field.ring:
```

Comment:
Shouldn't the `isinstance(g, PolyElement)` be built-in to the `__eq__` of `g.ring`?

Reply:
If `g` is not a `PolyElement` then it might not have a `ring` attribute.